schemas/groups/sources.json

```diff
@@ -4,28 +4,90 @@
   "id": "https://signalk.org/specification/0.9.0-SNAPSHOT/schemas/groups/sources.json#",
   "title": "sources",
   "description": "Metadata about the sources, eg. buses and connected sensors",
-  "patternProperties": {
-    ".*": {
+  "definitions":{
+    "sourcesBranch": {
       "type": "object",
       "patternProperties": {
-        ".*": {
-          "anyOf": [
+        "^((?!^0183$|^n2k$|^ais$|^i2c$|^other$|^label$|^type$|^manufacturerName$|^productName$|^softwareVersion$|^hardwareVersion$|^serialNumber$|^installationNote[1-2]$|^manufacturerInfo$)[A-Za-z0-9-])+$": {
+          "description": "This regex pattern is used for validation of the path of the source",
+          "oneOf": [
             {
+              "$ref": "#/definitions/sourcesBranch",
+              "example": "ais-b"
+            },
+            {
+              "$ref": "#/definitions/sourceObject"
+            }
+          ]
+        }
+      },
+      "additionalProperties": false
+    },
+    "sourceObject":{
+      "type": "object",
+
+      "patternProperties": {
+        "^((?!^0183$|^n2k$|^ais$|^i2c$|^other$|^label$|^type$|^manufacturerName$|^productName$|^softwareVersion$|^hardwareVersion$|^serialNumber$|^installationNote[1-2]$|^manufacturerInfo$)[A-Za-z0-9-])+$": {
+          "type": "object",
+          "oneOf": [
+            {
+              "required": [
+                "ais"
```

Comment:
I still don't understand how AIS is a type of source on its own. If i receive AIS data via 0183 or N2K, what is the source? If AIS why not 0183/N2K? 

Below an AIS source has message type property - what goes there? The type of the last message received? Why?

Or am i missing something here completely?

Reply:
Not my change, so I may not have the justification correct, but AIS could be received by a Signal K host directly via SDR or something. Also, giving them a source type of NMEA* sort of obscures their true origination. 0183, n2k, i2c, etc all imply a "local origination point". Whereas AIS is generally of a "non-local origination".

That said, this doesn't really belong in this PR. If we want to discuss adding AIS as a distinct source, that is it's own question.

Reply:
Ais stuff actually predates this PR, the changes seem to be related to formatting here. Just goes to show that autoformatting in a precommit hook or (enforced) by CI would make things simpler.

So this is not a stopper for this PR, a separate issue.